File: app/src/main/java/com/lightsoft/parceler/model/Employee.java
package com.lightsoft.parceler.model;

import org.parceler.Parcel;
import org.parceler.ParcelConstructor;

import java.util.List;

@Parcel(Parcel.Serialization.BEAN)
public class Employee {
    private String name;
    private String lastName;
    private List<Device> devices;

    @ParcelConstructor
    public Employee(String name, String lastName,
            List<Device> devices) {
        this.name = name;
        this.lastName = lastName;
        this.devices = devices;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    public List<Device> getDevices() {
        return devices;
    }

    public void setDevices(List<Device> devices) {
        this.devices = devices;
    }
}

File: kotlinparceler/src/main/java/com/lightsoft/kotlinparceler/model/Employee.kt
package com.lightsoft.kotlinparceler.model

import org.parceler.Parcel
import org.parceler.ParcelConstructor
import org.parceler.ParcelProperty

@Parcel(Parcel.Serialization.BEAN)
class Employee
@ParcelConstructor
constructor(@ParcelProperty("name") var name: String?,
            @ParcelProperty("lastName") var lastName: String?,
            @ParcelProperty("devices") var devices: ArrayList<Device>?) {

    constructor() : this(null, null, null)
}
